modules/apps/headless-apio/workflow/workflow-apio-impl/src/main/java/com/liferay/workflow/apio/internal/architect/resource/WorkflowTaskNestedCollectionResource.java

```diff
@@ -0,0 +1,174 @@
+import com.liferay.comment.apio.architect.identifier.CommentIdentifier;
+import com.liferay.person.apio.architect.identifier.PersonIdentifier;
+import com.liferay.portal.kernel.model.Company;
+import com.liferay.portal.kernel.model.User;
+import com.liferay.portal.kernel.service.UserLocalService;
+import com.liferay.portal.kernel.util.GetterUtil;
+import com.liferay.portal.kernel.workflow.WorkflowException;
+import com.liferay.portal.kernel.workflow.WorkflowTask;
+import com.liferay.portal.kernel.workflow.WorkflowTaskManager;
+import com.liferay.workflow.apio.architect.identifier.WorkflowTaskIdentifier;
+
+import java.io.Serializable;
+
+import java.util.List;
+import java.util.Map;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * Provides the information necessary to expose a WorkflowTask resources through a web API.
+ * The resources are mapped from the internal model {@link WorkflowTask}.
+ *
+ * @author Sarai Díaz
+ */
+@Component
+public class WorkflowTaskNestedCollectionResource
+	implements NestedCollectionResource
+		<WorkflowTask, Long, WorkflowTaskIdentifier, Long, PersonIdentifier> {
+
+	@Override
+	public NestedCollectionRoutes<WorkflowTask, Long, Long> collectionRoutes(
+		NestedCollectionRoutes.Builder<WorkflowTask, Long, Long> builder) {
+
+		return builder.addGetter(
+			this::_getPageItems, Company.class
+		).build();
+	}
+
+	@Override
+	public String getName() {
+		return "workflow-tasks";
+	}
+
+	@Override
+	public ItemRoutes<WorkflowTask, Long> itemRoutes(
+		ItemRoutes.Builder<WorkflowTask, Long> builder) {
+
+		return builder.addGetter(
+			this::_getWorkflowTask, Company.class
+		).build();
+	}
+
+	@Override
+	public Representor<WorkflowTask> representor(
+		Representor.Builder<WorkflowTask, Long> builder) {
+
+		return builder.types(
+			"WorkflowTask"
+		).identifier(
+			WorkflowTask::getWorkflowTaskId
+		).addBidirectionalModel(
+			"assignee", "tasks", PersonIdentifier.class,
+			WorkflowTask::getAssigneeUserId
+		).addBoolean(
+			"completed", WorkflowTask::isCompleted
+		).addDate(
+			"dateCompleted", WorkflowTask::getCompletionDate
+		).addDate(
+			"dateCreated", WorkflowTask::getCreateDate
+		).addDate(
+			"expires", WorkflowTask::getDueDate
+		).addLinkedModel(
+			"blogPost", BlogPostingIdentifier.class, this::_getBlogsEntryId
+		).addLinkedModel(
+			"comment", CommentIdentifier.class, this::_getCommentId
+		).addString(
+			"definitionName", WorkflowTask::getWorkflowDefinitionName
+		).addString(
+			"description", WorkflowTask::getDescription
+		).addString(
+			"name", WorkflowTask::getName
+		).addStringList(
+			"transitions", this::_getTaskTransitionsNames
+		).build();
+	}
+
+	private Long _getBlogsEntryId(WorkflowTask workflowTask) {
+		return GetterUtil.getLong(_getEntryClassPK(workflowTask));
+	}
+
+	private Long _getCommentId(WorkflowTask workflowTask) {
```

Comment:
this is strange

Reply:
i can use the same method _getAssetEntryClassPK instead of _getBlogsEntryId and _getCommentId, ok 👍